### PR TITLE
Issue 3156: Document fix: Segment store

### LINKED ISSUE #3156: Segment-store-service.md( few more diagrams included)
**Problem description**
1. New diagrams included
     a. Segment-store-synctier2.png
     b. Segment-store-container-startup.png
2. Content revised

### PR DESCRIPTION
**Pull request by Aparnarr**
Signed-off-by: Aparna <Aparna.Rr@emc.com>

**Change log description**  
1. 2 more diagrams included:
    (segment-store-synctier2.png and segment-store-recovery.png)
2. Minor edits
3. TOC added

**Purpose of the change**  
Fixes #3156

**How to verify it**  
(Optional: steps to verify that the changes are effective)

### CODE REVIEW

**Comment by Aparnarr** [comment]
segment-store-sync-tire2
segment-store-recovery

**Comment by Aparnarr** [comment]
@andreipaduroiu @ fpj requested for review

File: documentation/src/docs/segment-store-service.md [hunk 1]
```diff
@@ -66,117 +96,116 @@ Segment Containers are a logical grouping of Segments, and are responsible for a
...
-- **Storage Writer**: Processes the durable log operations and applies them to Tier 2 storage (in the order in which they were received). This component is also the one that coalesces multiple operations together, for better back-end throughput.
+- **Storage Writer**: Processes the durable log operations and applies them to Tier 2 Storage (in the order in which they were received). This component is also the one that coalesces multiple operations together, for better back-end throughput.
 
 ### Segment Container Metadata
-The Segment Container Metadata is critical to the good functioning and synchronization of its components. This metadata is shared across all components and it comes at two levels: "Container-wide Metadata" and "per-Segment Metadata". Each serves a different purpose and is described below.
+The Segment Container Metadata is critical to the good functioning and synchronization of its components. This metadata is shared across all components and it comes at two levels: "Container-wide metadata" and "per-Segment metadata". Each serves a different purpose and described below.
```
Comment (andreipaduroiu) [inline documentation/src/docs/segment-store-service.md]: Add "is" back. `is described` is how it should be.

File: documentation/src/docs/segment-store-service.md [hunk 2]
```diff
@@ -66,117 +96,116 @@ Segment Containers are a logical grouping of Segments, and are responsible for a
...
 - **Epoch**: A number that is incremented every time a successful recovery (Container Start) happens. This value is durably incremented and stored as part of recovery and can be used for a number of cases (a good use is Tier 2 fencing for HDFS, which doesn't provide a good, native mechanism for that).
-- **Active Segment Metadata**: Information about each active Segment (see next section below). A Segment is active if it has had activity (read or write) recently and is currently loaded in memory. If a Segment is not used for a while, or if there are many Segments currently active, a Segment becomes inactive by having its outstanding metadata flushed to Tier 2 Storage and evicted from memory.
+- **Active Segment Metadata**: Provides information about each active Stream Segment. A Segment is active if it has had activity (read or write) recently and is currently loaded in memory. If a Stream Segment is idle for a while, or if there are many Stream Segments currently active, a Stream Segment becomes inactive by having its outstanding metadata flushed to Tier 2 Storage and evicted from memory.
```
Comment (andreipaduroiu) [inline documentation/src/docs/segment-store-service.md]: It doesn't "provide" information, it "keeps" information. Either change it to "Keeps" or leave it as it was before, which is consistent with the rest of the bullets in this section.

File: documentation/src/docs/segment-store-service.md [hunk 3]
```diff
@@ -66,117 +96,116 @@ Segment Containers are a logical grouping of Segments, and are responsible for a
...
 
-- `Name`: The name of the Segment.
-- `Id`: Internally assigned unique Segment Id. This is used to refer to Segments, which is preferred to the Name. This Id is sticky for the lifetime of the Segment, which means that even if the Segment becomes inactive, a future reactivation will have it mapped to the same Id.
-- `StartOffset` (also known as `TruncationOffset`): The lowest offset of the data that is available for reading. A non-truncated segment will have Start Offset equal to 0, while subsequent Truncate operations will increase (but never decrease) this number.
+- `Name`: The name of the Stream Segment.
+- `Id`: Internally assigned unique Stream Segment ID. This is used to refer to Stream Segments, which is preferred to the Name. This ID is used for the entire lifetime of the Stream Segment, which means that even if the Stream Segment becomes inactive, a future reactivation will have it mapped to the same ID.
+- `StartOffset` (also known as `TruncationOffset`): The lowest offset of the data that is available for reading. A non-truncated Stream Segment will have Start Offset equal to _0_, while subsequent Truncate Operations will increase (but never decreases) this number.
```
Comment (andreipaduroiu) [inline documentation/src/docs/segment-store-service.md]: "decreases" -> "decrease" . Leave it as it was. This is in conjunction with a plural subject, hence we need the plural form.

File: documentation/src/docs/segment-store-service.md [hunk 4]
```diff
@@ -66,117 +96,116 @@ Segment Containers are a logical grouping of Segments, and are responsible for a
...
+    - `UpdateAttributesOperation`: Updates any attributes on a Stream Segment.
+    - `MetadataCheckpoint`: Includes an entire snapshot of the metadata. This can be useful during recovery. This contains all metadata up to this point, which is a sufficient base for all operations after it.
 
 ### Durable Log
 
-The _Durable Log_ is the central component that handles all Log Operations. All Operations (which are created by the Container) are added to the _Durable Log_, which processes them in the order in which they were received. It is made up of a few other components, all of which are working towards a single goal of processing all incoming operations as quickly as possible, without compromising data integrity.
+The _Durable Log_ is the central component that handles all Log Operations. All operations (which are created by the Container) are added to the _Durable Log_, which processes them in the order in which they were received. It is made up of a few other components, all of which are working towards a single goal of processing all incoming operations as quickly as possible, without compromising data integrity.
```
Comment (andreipaduroiu) [inline documentation/src/docs/segment-store-service.md]: You are inconsistent in your use of capital letters for `operation`. It's either `Operation` or `operation`. We need to choose one and use the same casing throughout the document.

File: documentation/src/docs/segment-store-service.md (hunk 1 above)
Comment (Aparnarr) [reply documentation/src/docs/segment-store-service.md]: Reverted.(Accepted Suggestion)

File: documentation/src/docs/segment-store-service.md (hunk 2 above)
Comment (Aparnarr) [reply documentation/src/docs/segment-store-service.md]: Accepted Suggestion

File: documentation/src/docs/segment-store-service.md (hunk 3 above)
Comment (Aparnarr) [reply documentation/src/docs/segment-store-service.md]: Accepted suggestion

File: documentation/src/docs/segment-store-service.md (hunk 4 above)
Comment (Aparnarr) [reply documentation/src/docs/segment-store-service.md]: will be using 👍 Operation->operation, Metadata->metadata

File: documentation/src/docs/segment-store-service.md [hunk 5]
```diff
@@ -99,25 +99,25 @@ Segment Containers are a logical grouping of Segments, and are responsible for a
...
+The Segment Container Metadata is critical to the good functioning and synchronization of its components. This metadata is shared across all components and it comes at two levels: "Container-wide metadata" and "per-Segment metadata". Each serves a different purpose and is described below.
 
 #### Container Metadata
 
 Each **Segment Container** needs to keep some general-purpose metadata that affects all operations inside the container:
 
-- **Operation Sequence Number**: The largest sequence number assigned by the _Durable Log_. Every time a new operation is received and successfully processed by the _Durable Log_, this number is incremented (its value never decreases or otherwise roll back, even if an operation failed to be persisted).
+- **Operation Sequence Number**: The largest sequence number assigned by the _Durable Log_. Every time a new operation is received and successfully processed by the _Durable Log_, this number is incremented (its value never decrease or otherwise roll back, even if an operation failed to be persisted).
```
Comment (andreipaduroiu) [inline documentation/src/docs/segment-store-service.md]: `its value never decreases` . This is  a singular noun. Revert this change.

Comment (tkaitchuck) [reply documentation/src/docs/segment-store-service.md]: It should be `its value will never decrease`

Comment (Aparnarr) [reply documentation/src/docs/segment-store-service.md]: Accepted suggestion

**Comment by Aparnarr** [comment]
@andreipaduroiu @tkaitchuck any further comments??
